visualizations/frontend/filters/v2/TTagsInput.vue

```diff
@@ -1,145 +1,144 @@
-						if (key && value) {
-							items.push({
-								key: key.value,
-								value: value.value,
-							})
-						}
-					}
-				}
-				return items;
-			},
-			firstKey() {
-				// Used for single tag
-				if (this.tags.length) {
-					return this.tags[0].key;
-				}
-				
-				if (this.tempTags) {
-					return this.tempTags[0].key;
-				}
-				
-				return null;
-			}
-		},
-		methods: {
-			onVisualizationInit() {
-				const initial_value = this.getFilterValue("selected_items");
+      if (this.rows.length) {
+        for (let row of this.rows) {
+          const key = row[this.tKeyColumn];
+          const value = row[this.tValueColumn];
+          if (key && value) {
+            items.push({
+              key: key.value,
+              value: value.value,
+            });
+          }
+        }
+      }
+      return items;
+    },
+    firstKey() {
+      // Used for single tag
+      if (this.tags.length) {
+        return this.tags[0].key;
+      }
 
-				if (initial_value) {
-					const selectedTags = [];
-                    const tempTags = [];
-					const urlTags = initial_value.split('|');
-					
-					for (let item of urlTags) {
-						const splitPair = item.split('_');
-						const key = splitPair.length > 1 ? splitPair[1] : splitPair[0];
-						const value = splitPair[0];
+      if (this.tempTags) {
+        return this.tempTags[0].key;
+      }
 
-						if (key && value) {
-							if (this.isSingleTag) {
-								selectedTags.push(value);
-								tempTags.push({ key, value })
-							} else {
-								selectedTags.push({ key, value });
-							}
-						}
-					}
-					
-				    this.tempTags = tempTags;
-				    this.selectedTags = selectedTags;
-				}
-			},
-			handleUpdate(items) {
-				const values = [];
-				for (let item of items) {
-					if (this.isSingleTag) {
-						if (!this.firstKey) {
-							break;
-						}
-						values.push(`${item}_${this.firstKey}`);
-					} else {
-						values.push(`${item.value}_${item.key}`);
-					}
-				}
-				this.setFilterValue("selected_items", values.join('|'));
-			},
-			handleOpen() {
-				this.fetchLayerData();
-			},
-		},
-	}
+      return null;
+    },
+  },
+  methods: {
+    onVisualizationInit() {
+      const initial_value = this.getFilterValue("selected_items");
+      if (initial_value) {
+        try {
+          const urlTags = JSON.parse(initial_value);
+          for (const [key, value] of Object.entries(urlTags)) {
+            let tags = value.map((v) => {
+              return {
+                key,
+                value: v,
+              };
+            });
+            if (this.isSingleTag) {
+              // base-multiple-value-input is used for single tag filter.
+              this.tempTags = [...tags, ...this.tempTags]; // Without defaults base-multiple-value-input cannot show selected tags.
+              this.selectedTags = [...value, ...this.selectedTags]; // base-multiple-value-input needs exact value instead of key:val pairs.
```

Comment:
Is `value` normally a string? If yes, would the end result be what you expect I would expect the string to be split like so: 

```
const aValue = "some_value" 
const myArray = [...aValue] 
console.log(myArray)
Array(10) [ "s", "o", "m", "e", "_", "v", "a", "l", "u", "e" ]
```